app/services/abstract_document_service_registry.rb

```diff
@@ -72,6 +72,20 @@ def publish(document_id)
     )
   end
 
+  def republish(document_id)
+    RepublishDocumentService.new(
+      document_repository: document_repository,
+      listeners: observers.republication,
+      document_id: document_id,
+    )
+  end
+
+  def republish_all
```

Comment:
It seems odd to have this method perform the action when every other public method in this class returns a callable.